src/c2ffi/integration/cl-ppcre.lisp

```diff
@@ -0,0 +1,36 @@
+;;;; -*- Mode: lisp; indent-tabs-mode: nil -*-
+;;;
+;;; cl-ppcre.lisp --- Adds cl-ppcre support to name matching.
+;;;
+;;; Copyright (C) 2015, Attila Lendvai <attila@lendvai.name>
+;;;
+;;; Permission is hereby granted, free of charge, to any person
+;;; obtaining a copy of this software and associated documentation
+;;; files (the "Software"), to deal in the Software without
+;;; restriction, including without limitation the rights to use, copy,
+;;; modify, merge, publish, distribute, sublicense, and/or sell copies
+;;; of the Software, and to permit persons to whom the Software is
+;;; furnished to do so, subject to the following conditions:
+;;;
+;;; The above copyright notice and this permission notice shall be
+;;; included in all copies or substantial portions of the Software.
+;;;
+;;; THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+;;; EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+;;; MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+;;; NONINFRINGEMENT.  IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+;;; HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+;;; WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+;;; OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+;;; DEALINGS IN THE SOFTWARE.
+;;;
+
+(in-package #:cffi/c2ffi)
+
+(defun make-rule-matcher/cl-ppcre (pattern)
+  (let ((scanner (cl-ppcre:create-scanner pattern)))
+    (named-lambda cffi/c2ffi/cl-ppcre-rule-matcher
+        (string)
+      (funcall scanner string 0 (length string)))))
+
+(setf *make-matcher-from-rule-hook* 'make-rule-matcher/cl-ppcre)
```

Comment:
IIUC, the presence of this systems affects how the patterns are processed, right? Doesn't that mean that a system that was using plain strings might feel if for some reasons cffi/c2ffi+cl-ppcre is loaded?

Can't we just depend on cl-ppcre unconditionally and call it a day? (Hmm, does that mean we'd be requiring cl-ppcre just for loading a cffi/c2ffi-generated set of bindings? Can we avoid that somehow? More generally, can we avoid loading c2ffi altogether if the bindings are already generated? Hmm, this last bit probably doesn't make much sense. The idea is to cache [i.e., commit] specs, not the generated bindings, right?)